assets/src/hooks/usePersistedStateReducer.ts

```diff
@@ -27,40 +26,96 @@ const PERSISTED_KEYS: Key[] = [
 const usePersistedStateReducer = (): [State, Dispatch] => {
   const [state, dispatch] = useReducer(reducer, undefined, init)
 
-  const persistableState = filter(state, PERSISTED_KEYS)
+  const locallyPersistableState = filter(state, LOCALLY_PERSISTED_KEYS)
+
   useEffect(() => {
-    saveState(APP_STATE_KEY, persistableState)
-  }, [persistableState])
+    saveState(APP_STATE_KEY, locallyPersistableState)
+  }, [locallyPersistableState])
+
+  const { selectedRouteIds, ladderDirections, ladderCrowdingToggles } = state
+
+  const [firstLoadDone, setFirstLoadDone] = useState(false)
+
+  useEffect(() => {
+    if (firstLoadDone) {
+      putRouteSettings(state)
+    } else {
+      setFirstLoadDone(true)
+    }
+  }, [selectedRouteIds, ladderDirections, ladderCrowdingToggles])
```

Comment:
Ooh, I like this way of updating the backend more than the approach I did for usersettings of calling the api call in every place the settings change.
Do you think it'd be possible/easy to split this into like a `useUpdateBackendAboutRouteSettings()` hook and also refactor the user settings updating into a `useUpdateBackendAboutUserSettings()` hook?
(This PR is big and it might be better to do those in a followup PR)

Reply:
I do think that would be possible and worth doing, but I do also think that a followup is indicated, since as you say this has turned out to be a big PR.